pkg/tesla/fingerprint/fingerprint.go

```diff
@@ -2,22 +2,47 @@
 package fingerprint
 
 import (
+	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/DIMO-Network/model-garage/pkg/cloudevent"
+	"github.com/DIMO-Network/model-garage/pkg/tesla"
+	"github.com/teslamotors/fleet-telemetry/protos"
 	"github.com/tidwall/gjson"
+	"google.golang.org/protobuf/proto"
 )
 
 // DecodeFingerprintFromData decodes a fingerprint from the data portion of a CloudEvent.
-func DecodeFingerprintFromData(data []byte) (cloudevent.Fingerprint, error) {
+func DecodeFingerprintFromData(ce cloudevent.CloudEvent[json.RawMessage]) (cloudevent.Fingerprint, error) {
 	fingerPrint := cloudevent.Fingerprint{}
-	result := gjson.GetBytes(data, "vin")
-	if !result.Exists() {
-		return fingerPrint, fmt.Errorf("vin field not found")
-	}
-	if result.Type != gjson.String {
-		return fingerPrint, fmt.Errorf("vin field is not a string")
+	switch ce.DataVersion {
+	case tesla.FleetTelemetryDataVersion:
+		var payloads tesla.TelemetryData
+		if err := json.Unmarshal(ce.Data, &payloads); err != nil {
+			return fingerPrint, err
```

Comment:
nitpick: We parse twice here, from JSON and Protobuf. It might be worth wrapping the resulting errors so that the origin of any parsing error that bubbles up from this is clearer.